_posts/2019-06-18-list-strategy.md

```diff
@@ -0,0 +1,101 @@
+This mechanism was quite powerful and widely used, but had some drawbacks:
+
+- Bazel's defaults did not account for remote execution at all, therefore we had to make sure to set the execution strategy for every mnemonic manually.
+
+- Starlark rules providing a persistent worker had to ship a .bazelrc file that sets `--strategy=Mnemonic=worker` (e.g. look at [rules_scala](https://github.com/bazelbuild/rules_scala#getting-started), where `--strategy=Scalac=worker` is required in order to run it with a persistent worker). 
+
+- Bazel has hardcoded defaults for native action mnemonics. For example, Bazel will use sandboxing (if it's available) for all actions by default but has a hardcoded default to use persistent workers for Java compilation actions.
+
+- To configure fallback strategy one had to provide additional flag `--remote_local_fallback_strategy=`.
+
+> ### Example
+> Let's configure a build to run remotely and fallback onto `local` strategy in case remote execution is not possible. The configuration would be something like:
+> 
+``` 
+$ bazel build
+ --spawn_strategy=remote
+ --strategy=Javac=remote
+ --strategy=Closure=remote
+ --strategy=Genrule=remote
+ --remote_local_fallback_strategy=local
+ ... 
+```
+
+## After Bazel 0.27
+- Bazel now *auto-detects* the execution strategy, if no strategy flag is provided.  
+If none of the strategy flags was used, bazel will generate a default list of strategies `remote,worker,sandboxed,local` and, for every action it wants to execute, will pick up the first strategy that can execute it.
+
+- The user can pass a comma-separated list of strategies to the above mentioned flags: `--spawn_strategy=remote,worker,linux-sandbox`.  
+Each strategy now knows whether it can execute a given action.
+For any action that it wants to execute, Bazel just picks the first strategy from the given list that can execute the action. 
+
+- If an action cannot be executed with any of the given strategies, the build will fail.  
+If any of the strategy flags is provided, then Bazel will use ONLY those strategies that are listed there. 
+Which means that `--spawn_strategy=remote` will NOT fallback to a `sandboxed` strategy or to any other strategy in case remote action failed or is not possible at all.
+Moreover, `--remote_local_fallback_strategy=local` is now deprecated and made no-op. 
+
+### Why is it better?
+- Now you can control your execution strategy and have your build fail if any non-sandboxable or non-remotable action sneaks in! 
+More reproducibility and safety for your builds!
+- The strategies no longer do their own custom fallback, simplifying the code and unifying the behavior.
+- You might even completely forget about strategies configurations if the default behavior satisfies your needs.
+
+> ### Example
+> This is how the previous example will look like now:
```

Comment:
Quotations do not look awesome here, because they are broken by verbatim multi-line blocks :( ....
Try inline code snippets maybe?  \`...\` instead of \`\`\`\...\`\`\` - your code snippets are not that long anyway

Reply:
IMO it looks fine: 
![example-1](https://user-images.githubusercontent.com/11422276/59752238-486cbb80-9282-11e9-9e7c-49d0f6538dfc.png)

wdyt?

Reply:
aaa, sorry. you were talking about the second example.

Reply:
Well, actually the second one is also not bad. Still don't like this?

![example-2](https://user-images.githubusercontent.com/11422276/59752618-e3659580-9282-11e9-9263-20d24f928d69.png)

Reply:
Ah nice, I was looking at how GitHub renders this - that is nice